pkg/build/nodeimage/internal/container/docker/archive.go

```diff
@@ -117,6 +117,15 @@ func EditArchiveRepositories(reader io.Reader, writer io.Writer, editRepositorie
 				return err
 			}
 			hdr.Size = int64(len(b))
+			// edit image config when we find that
+		} else if strings.HasSuffix(hdr.Name, ".json") {
```

Comment:
this is very obscure, so every json but manifest.json is an image config manifest?

Reply:
yes, they will be named `<hash>.json`. we could get the names from the manifest.json, but then we need to do two passes over the tar or depend on ordering or ...

we could do this, but it doesn't seem necessary. images following the spac should only contain `manifest.json` and `<hash>.json` .json files where the latter are image config files.

links to the spec are in the comments above https://github.com/moby/moby/blob/master/image/spec/v1.2.md

Reply:
a typical image archive only has two .json files in it, but it's possible to create a tarball with multiple images, in which case there is one `manifest.json` and one config hash.json. 

The manifest contains an array of structs each with the config file name for an image, it's tags, and it's layers. There is also the legacy repositories file for the tags.

The config file specifies the non-layer data like architecture/os, entrypoint, env, etc. for one image. We are patching the architecture field to avoid kind only working on highly patched kubernetes builds.

Reply:
That is  a lot of information, do you think future us can figure it out?

Reply:
There are comments and there are links to the spec, which is not very lengthy :-)

Reply:
:+1: